keel-echo/src/main/kotlin/com/netflix/spinnaker/keel/echo/ManualJudgementNotifier.kt

```diff
@@ -0,0 +1,73 @@
+import com.netflix.spinnaker.keel.echo.model.EchoNotification
+import com.netflix.spinnaker.keel.events.ConstraintStateChanged
+import kotlinx.coroutines.runBlocking
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Component
+
+@Component
+/**
+ * Listens for [ConstraintStateChanged] events where the constraint is a [ManualJudgementConstraint] and sends
+ * out notifications so that users can take action.
+ */
+class ManualJudgementNotifier(
+  private val echoService: EchoService
+) {
+
+  @EventListener(ConstraintStateChanged::class)
+  fun constraintStateChanged(event: ConstraintStateChanged) {
+    // if this is the first time the constraint was evaluated, send a notification
+    // so the user can react via other interfaces outside the UI (e.g. e-mail, Slack)
+    if (event.constraint is ManualJudgementConstraint &&
+      event.previousState == null &&
+      event.currentState.status == ConstraintStatus.PENDING) {
+      event.environment.notifications.map {
+        // TODO: run in parallel
+        runBlocking {
+          echoService.sendNotification(
+            EchoNotification(
+              notificationType = EchoNotification.Type.valueOf(it.type.value().toUpperCase()),
+              to = listOf(it.address),
+              // templateGroup = TODO
+              severity = EchoNotification.Severity.NORMAL,
+              source = EchoNotification.Source(
+                // FIXME: Environment should probably have a reference to the application name...
+                application = event.environment.resources.firstOrNull()?.application
+                // TODO: anything for executionType, executionId, user?
+              ),
+              additionalContext = mapOf(
+                "formatter" to "MARKDOWN",
+                "subject" to "Manual artifact promotion approval",
+                "body" to
+                  ":warning: The artifact *${event.currentState.artifactVersion}* from delivery config " +
+                  "*${event.currentState.deliveryConfigName}* requires your manual approval for deployment " +
+                  "into the *${event.currentState.environmentName}* environment."
+              ),
+              interactiveActions = EchoNotification.InteractiveActions(
+                callbackServiceId = "keel",
+                callbackMessageId = event.currentState.uid ?: error("ConstraintState.uid not present"),
+                actions = listOf(
+                  EchoNotification.ButtonAction(
+                    name = "manual-judgement",
+                    label = "Approve",
+                    value = ConstraintStatus.OVERRIDE_PASS.name
+                  ),
+                  EchoNotification.ButtonAction(
+                    name = "manual-judgement",
+                    label = "Reject",
+                    value = ConstraintStatus.OVERRIDE_FAIL.name
+                  )
+                ),
+                color = "#fcba03"
+              )
```

Comment:
The magic pretty much boils down to this. This is what tells echo to include the buttons in the Slack message and provides callback properties that allow echo to call us back with the response.